code/go/0chain.net/blobbercore/allocation/copyfilechange.go

```diff
@@ -132,34 +125,38 @@ func (rf *CopyFileChange) processCopyRefs(ctx context.Context, affectedRef, dest
 		newRef.LookupHash = reference.GetReferenceLookup(newRef.AllocationID, newRef.Path)
 		newRef.Attributes = datatypes.JSON(string(affectedRef.Attributes))
 		destRef.AddChild(newRef)
+
 		for _, childRef := range affectedRef.Children {
 			rf.processCopyRefs(ctx, childRef, newRef, allocationRoot)
 		}
-	} else {
-		newFile := reference.NewFileRef()
-		newFile.ActualFileHash = affectedRef.ActualFileHash
-		newFile.ActualFileSize = affectedRef.ActualFileSize
-		newFile.AllocationID = affectedRef.AllocationID
-		newFile.ContentHash = affectedRef.ContentHash
-		newFile.CustomMeta = affectedRef.CustomMeta
-		newFile.MerkleRoot = affectedRef.MerkleRoot
-		newFile.Name = affectedRef.Name
-		newFile.ParentPath = destRef.Path
-		newFile.Path = filepath.Join(destRef.Path, affectedRef.Name)
-		newFile.LookupHash = reference.GetReferenceLookup(newFile.AllocationID, newFile.Path)
-		newFile.Size = affectedRef.Size
-		newFile.MimeType = affectedRef.MimeType
-		newFile.WriteMarker = allocationRoot
-		newFile.ThumbnailHash = affectedRef.ThumbnailHash
-		newFile.ThumbnailSize = affectedRef.ThumbnailSize
-		newFile.ActualThumbnailHash = affectedRef.ActualThumbnailHash
-		newFile.ActualThumbnailSize = affectedRef.ActualThumbnailSize
-		newFile.EncryptedKey = affectedRef.EncryptedKey
-		newFile.Attributes = datatypes.JSON(string(affectedRef.Attributes))
-		newFile.ChunkSize = affectedRef.ChunkSize
-
-		destRef.AddChild(newFile)
 	}
+
+	newFile := reference.NewFileRef()
```

Comment:
If the affectedRef is a DIR, the code above would add it to the destRef as a directory, do we need to add it as a file again here

Reply:
fixed